### ISSUE TITLE
ConnectBot Can´t find key Files

### ISSUE THREAD

**Issue by JPRuehmann** [connectbot/connectbot#484]
Where I must put the Key Files?
ConnectBot can´t find them anywhere.
I had puted them in Downloads.
Thanks,

**Comment by kruton** [comment]
What version of Android are you using? What version of ConnectBot?

**Comment by JPRuehmann** [comment]
Android 5.1.1
ConnectBot 1.9.2

**Comment by f00bar12** [comment]
When you select Manage Pubkeys it should take you to the pubkeys that you currently have loaded.  In your case it will probably be empty, then in the upper right hand corner there should be a folder icon, click that and it will give you a menu with different options to load a key from.  Select Downloads if that's where you've put your key, then click on your key.  It should load it into the previous screen and it can now be used in your connections.

**Comment by JPRuehmann** [comment]
Am 04.03.2017 um 01:36 schrieb
      f00bar12:
    
    
      When you select Manage Pubkeys it should take you to the
        pubkeys that you currently have loaded. In your case it will
        probably be empty, then in the upper right hand corner there
        should be a folder icon, click that and it will give you a menu
        with different options to load a key from. Select Downloads if
        that's where you've put your key,
    
    To this Point everything works but the Keyfiles won´t shgow up there
    like in all other filemanagers and d other ssh-clients find them
    just fine only connectBot will find nothing.
    
       then click on your key. It should load it into the previous
        screen and it can now be used in your connections.
    
    I can´t because he won´t show up.
    
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/connectbot/connectbot","title":"connectbot/connectbot","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/connectbot/connectbot"}},"updates":{"snippets":[{"icon":"PERSON","message":"@f00bar12 in #484: When you select Manage Pubkeys it should take you to the pubkeys that you currently have loaded.  In your case it will probably be empty, then in the upper right hand corner there should be a folder icon, click that and it will give you a menu with different options to load a key from.  Select Downloads if that's where you've put your key, then click on your key.  It should load it into the previous screen and it can now be used in your connections."}],"action":{"name":"View Issue","url":"https://github.com/connectbot/connectbot/issues/484#issuecomment-284108888"}}}
    
    Thanks,
    -- 
      -=============================== Jan-Peter Rühmann & Kuma ===============================-
 Gubkower Str.7               [  Tel.:  +49 (38205) 65484  ]       jan-Peter@ruehmann.name
 18195 Prangendorf            [  FAX:   +49 (38205) 65212  ]      http://www.ruehmann.name
                              [  Tel.:  +49 (38205) 65215  ]
                              [  Mobil: +49 (162) 1316054  ]           IT-Servicetechniker
Skype: jan-peter_ruehmann / ICQ: 288192920 / WhatsApp: 491621316054 / Twitter: @JPRuehmann
-========================================================================================-
                  Die Verwendung der Daten zu Werbezwecken ist verboten.

**Comment by f00bar12** [comment]
Can you take a screenshot of the pubkeys screen and then the screen you get when you click on the folder icon and post it here?

**Comment by JPRuehmann** [comment]
Here
Hope that helps.
Thanks
![screenshot_2017-03-06-12-02-29](https://cloud.githubusercontent.com/assets/4321829/23607831/c272386e-0266-11e7-8f6f-79bf40b6957e.png)
![screenshot_2017-03-06-12-04-40](https://cloud.githubusercontent.com/assets/4321829/23607832/c272480e-0266-11e7-940a-553162d56c99.png)

**Comment by ngoonee** [comment]
I'm having the same problem. The file dialog which connectbot brings up only sees documents, images, videos, and music. It doesn't see any other types of files (in fact it can't even browse anywhere, I only see 'Recents', some Google Drive links, and surprisingly a termux link...

**Comment by Drugoy** [comment]
Looks like it is impossible to import the key unless you download it via a browser, but that's a pretty hard task, because keys are simple text files and most browsers will just show their contents instead of downloading them as file.

**Comment by PhilHudson** [comment]
Even downloading from a web server doesn't work for me (Android 5.1.1). The stock Android browser renames the private key, adding a .bin suffix, and after import it doesn't connect to my remote host even with the key unlocked and the corresponding public key added to that host's authorized_keys file. Opera Mini doesn't add its downloads to whatever OS-level "Downloads" list (virtual folder, whatever) Connectbot is using. Bluetooth sending from my PC doesn't get added to the list/virtual-folder either. This is a really bad bug. Please up the severity and/or priority, guys, and/or provide us with a workaround.

**Comment by bencaradocdavies** [comment]
I reported my workaround here: https://github.com/connectbot/connectbot/issues/541

tl;dr: use Chrome to "Download link", works with local files

Copying local `id_rsa` to the "Download" folder does not work because it does not appear in the "Downloads" virtual folder, but I found a workaround: I used Chrome to navigate to `file:///storage/emulated/0/foldername` (and once deep enough I can follow links), long-pressed on `id_rsa` and selected "Download link". `id_rsa` is copied to the  "Download" but now appears in the "Downloads" virtual folder and can be added in "Manage pubkeys" in ConnectBot. Horrible, but it works. I am using Marshmallow.

**Comment by GL0RI0U5** [comment]
Hello please does anyone know where connectbot saves downloaded files?.
I tried to scrap a website so I can get the source code and connectbot says it's saved as index.html but I don't know where to find index.html

**Comment by nindanaoto** [comment]
This seems to be outdated, since the key import implemented in #1688 is working.

**Comment by Drugoy** [comment]
Can confirm. @kruton please, close this ticket.
